mne/realtime/lsl_client.py

```diff
@@ -0,0 +1,117 @@
+    tmax : float
+        Time instant to stop receiving buffers.
+    buffer_size : int
+        Size of each buffer in terms of number of samples.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+    """
+
+    def __init__(self, host, port=None, wait_max=10., tmin=None,
+                 tmax=np.inf, buffer_size=1000, verbose=None):  # noqa: D102
+        super(LSLClient, self).__init__(host, port, tmin, tmax,
+                                        buffer_size, wait_max, verbose)
+
+    @fill_doc
+    def get_data_as_epoch(self, n_samples=1024, picks=None):
+        """Return last n_samples from current time.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples to fetch.
+        %(picks_all)s
+
+        Returns
+        -------
+        epoch : instance of Epochs
+            The samples fetched as an Epochs object.
+
+        See Also
+        --------
+        mne.Epochs.iter_evoked
+        """
+        wait_time = n_samples * 5. / self.info['sfreq']
+
+        # create an event at the start of the data collection
+        events = np.expand_dims(np.array([0, 1, 1]), axis=0)
+        samples, _ = self.client.pull_chunk(max_samples=n_samples,
+                                            timeout=wait_time)
+        data = np.vstack(samples).T
+
+        picks = _picks_to_idx(self.info, picks, 'all', exclude=())
+        info = pick_info(self.info, picks)
+        return EpochsArray(data[picks][np.newaxis], info, events)
+
+    def iter_raw_buffers(self):
+        """Return an iterator over raw buffers."""
+        pylsl = _check_pylsl_installed(strict=True)
+        inlet = pylsl.StreamInlet(self.client)
+
+        while True:
+            samples, _ = inlet.pull_chunk(max_samples=self.buffer_size)
+
+            yield np.vstack(samples).T
+
+    def _connect(self):
+        pylsl = _check_pylsl_installed(strict=True)
+        stream_info = pylsl.resolve_byprop('source_id', self.host,
+                                           timeout=1)[0]
+        self.client = pylsl.StreamInlet(info=stream_info,
+                                        max_buflen=self.buffer_size)
+
+        return self
+
+    def _create_info(self):
+        sfreq = self.client.info().nominal_srate()
+
+        lsl_info = self.client.info()
+        ch_info = lsl_info.desc().child("channels").child("channel")
+        ch_names = list()
+        ch_types = list()
+        ch_type = lsl_info.type()
+        for k in range(1,  lsl_info.channel_count() + 1):
+            ch_names.append(ch_info.child_value("label") or
+                            '{} {:03d}'.format(ch_type.upper(), k))
+            ch_types.append(ch_info.child_value("type") or
+                            ch_type.lower())
+            ch_info = ch_info.next_sibling()
+
+        info = create_info(ch_names, sfreq, ch_types)
```

Comment:
so for instance @teonbrooks you don't have the channel locations here. Could you plot a topomap or something with data from this client?

Reply:
this would require that the LSL stream has standard 10-20 or 10-10 names. if so, we could use the standard montage and create the topo-oriented plots. we can make that a follow-on PR for expanded functionality. i think it would be worth having.

Reply:
okay, that sounds good. I am fine with adding the functionality in small PRs

Reply:
actually, this is easier to implement than I thought. Included `standard_1005` as the default if the channel type is EEG